sdk/cosmos/azure_data_cosmos/README.md

```diff
@@ -1,5 +1,200 @@
+
+## Getting started
+
+### Install the package
+
+Install the Azure Cosmos DB SDK for Rust with cargo:
+
+```bash
+cargo add azure_data_cosmos
+```
+
+### Prerequisites
+
+* An [Azure subscription] or free Azure Cosmos DB trial account.
+
+Note: If you don't have an Azure subscription, create a free account before you begin.
+You can Try Azure Cosmos DB for free without an Azure subscription, free of charge and commitments, or create an Azure Cosmos DB free tier account, with the first 400 RU/s and 5 GB of storage for free. You can also use the Azure Cosmos DB Emulator with a URI of https://localhost:8081. For the key to use with the emulator, see [how to develop with the emulator](https://learn.microsoft.com/azure/cosmos-db/how-to-develop-emulator).
+
+### Create an Azure Cosmos DB account
+
+You can create an Azure Cosmos DB account using:
+
+* [Azure Portal](https://portal.azure.com).
+* [Azure CLI](https://learn.microsoft.com/cli/azure).
+* [Azure ARM](https://learn.microsoft.com/azure/cosmos-db/quick-create-template).
+
+#### Authenticate the client
+
+In order to interact with the Azure Cosmos DB service you'll need to create an instance of the `CosmosClient` struct. To make this possible you will need a URL and key of the Azure Cosmos DB service.
+
+## Examples
+
+The following section provides several code snippets covering some of the most common Azure Cosmos DB NoSQL API tasks, including:
+* [Create Client](#create-cosmos-db-client "Create Cosmos DB client")
+* [Create Database](#create-database "Create Database")
+* [Create Container](#create-container "Create Container")
+* [CRUD operation on Items](#crud-operation-on-items "CRUD operation on Items")
+
+### Create Cosmos DB Client
+
+The clients support different forms of authentication. The Azure Cosmos DB SDK for Rust supports authorization via Microsoft Entra identities or an account key. However, support for account keys is only available when the `key_auth` feature is enabled in your Cargo.toml file.
+
+```toml
+[dependencies]
+azure_data_cosmos = { version = "...", features = ["key_auth"] }
+```
+
+We **strongly** recommend using Microsoft Entra ID for authentication, rather than account keys.
+
+**Using Microsoft Entra ID**
+
+```rust
+use azure_identity::DefaultAzureCredential;
+use azure_data_cosmos::CosmosClient;
+
+async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let cosmos_client = CosmosClient::new("myAccountEndpointURL", credential, None)?;
+    Ok(())
+}
+```
+
+**Using account keys**
+
+```rust
+use azure_core::credentials::Secret;
+use azure_data_cosmos::CosmosClient;
+
+const COSMOS_DB_ENDPOINT: &str = "someEndpoint";
+const COSMOS_DB_KEY: &str = "someKey";
+
+async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    let key = Secret::from(COSMOS_DB_KEY);
+    let cosmos_client = CosmosClient::with_key(COSMOS_DB_ENDPOINT, key, None)?;
+    Ok(())
+}
+```
+
+### Create Database
```

Comment:
We've been creating a TOC of bookmarks, though I'm not particular fond of this (hard to keep maintained, and GitHub does it anyway). @ronniegeraghty are ya'll trying to standardize on that as well?

Reply:
The ToC at the Top of the Examples section? Not something I'm deeply attached to. My though was it would help readers scan through the available examples quickly without having to scroll past some of our longer code blocks.

Reply:
Given GitHub generates a TOC if they're interested (granted, not the most visible - see below), I'd prefer to nix it rather than keep it up to date. I already forgot to add one heading this week.

![image](https://github.com/user-attachments/assets/a5ea822a-9698-467d-bb65-7ccbe7e2b364)